deployer/src/deployer/upload.py

```diff
@@ -583,6 +600,12 @@ def on_task_complete(task):
         # now deleted.
         now = datetime.datetime.utcnow().replace(tzinfo=UTC)
         delete_keys = []
+
+        archived_files_as_keys = set()
+        for file in archived_files:
```

Comment:
Again, assuming we make `parse_archived_txt_file` a generator:
```suggestion
        for file in parse_archived_txt_file(archived_txt_file) if archived_txt_file else ():
```